drivers/soundwire/intel.c

```diff
@@ -1513,6 +1513,7 @@ static int intel_link_probe(struct auxiliary_device *auxdev,
 
 	bus->link_id = auxdev->id;
 	bus->dev_num_ida_min = INTEL_DEV_NUM_IDA_MIN;
+	bus->clk_stop_timeout = 20;
```

Comment:
Suggested edit to the commit message.

the bus->clk_stop_timeout member is only initialized to a non-zero value during the codec driver probe. This can lead to corner cases where this value remains pegged at zero when the bus suspends, which results in an endless loop in  sdw_bus_wait_for_clk_prep_deprep().

Corner cases include configurations with no codecs described in the firmware, or delays in probing codec drivers. 

Initializing the default timeout to the smallest non-zero value avoid this problem and allows for the existing logic to be preserved: the bus->clk_stop_timeout is set as the maximum required by all codecs connected on the bus.

Reply:
I'm happy to do this change; though it makes me wonder if the better solution wouldn't be to clamp the retry to 1 in _prep_deprep rather then  fixing this initialization